execution/block-partitioner/src/sharded_block_partitioner/partitioning_shard.rs

```diff
@@ -42,6 +43,7 @@ impl PartitioningShard {
             control_rx,
             result_tx,
             cross_shard_client,
+            num_rounds_finished: 0,
```

Comment:
better leave this partioning shard state less and not store round number here, can you pass it in the partitioning command itself?

Reply:
sure, but it looks to me partitioning shards are already stateful? Fields `shard_id` and the rxs/txs are all tied to a partitioning session already.

Reply:
fixed